cirq-core/cirq/study/resolver.py

```diff
@@ -63,7 +64,9 @@ def __init__(self, param_dict: 'cirq.ParamResolverOrSimilarType' = None) -> None
             return  # Already initialized. Got wrapped as part of the __new__.
 
         self._param_hash: Optional[int] = None
-        self.param_dict = cast(ParamDictType, {} if param_dict is None else param_dict)
+        self.param_dict = cast(
+            ParamDictType, {} if param_dict is None else copy.copy(cast(Any, param_dict))
```

Comment:
Instead of `copy.copy` I'd prefer to just create a new dict:
```suggestion
            ParamDictType, {} if param_dict is None else dict(param_dict)
```
Generally I try to avoid the `copy` module because it feels too magic. But others may disagree.